### DISCUSSION TITLE
Feature request: Update list of supported locales on the fly (i18n on demand)

### DISCUSSION THREAD

**Discussion by 7iomka** [vercel/next.js#18330] [Ideas]
# Feature request
I read new Next.js feature - https://nextjs.org/docs/advanced-features/i18n-routing
I see that list of locales is hardcoded inside next.config.js file.
That means that if I add from my admin panel a new locale (part of app that works through REST Api) next will not receive it, until next server will be restarted and manually changed list in `i18n.locales` array.
How to deal with maintaining locales updates on demand?
Any ideas?
Thanks!

**Comment by timneutkens** [comment]
> How to deal with maintaining locales updates on demand?

This is not supported currently. We're currently not planning to add support for it. If there's enough interest for this we can re-evaluate though.
